### DISCUSSION TITLE
Tilt caused by SRS within C-band signals

### DISCUSSION THREAD

**Discussion by caffery-chen** [Telecominfraproject/oopt-gnpy#393] [Q&A] [answered]
I would love to ask whether GNPY consider the power tilt caused by SRS from C-band short wavelength to longer wavelength?
This might alter the NLI during propagation.

**Comment by jktjkt** [comment, marked as answer]
In the current release this is implemented if you instantiate a `RamanFiber` instead of a `Fiber` in the network topology. This "pumpless" power transfer should be properly accounted for.

@AndreaDAmico , if I recall correctly your recent work will make this the default, right?

**Reply by AndreaDAmico** [reply]
In my recent work, the SRS can be taken into account also in the `Fiber` instance, but it is not the default. In order to enable this feature, the Raman flag within the simulation parameters has to be set as `True`.

**Reply by caffery-chen** [reply]
Dear @AndreaDAmico @jktjkt,

in order to do as Andrea suggested, I need to pass the sim_param.json to the transmission_data_example. 

Should I also remove/modify the 
  "nli_parameters": {
  	"nli_method_name": "ggn_spectrally_separated",
  	"wdm_grid_size": 50e9,
  	"dispersion_tolerance": 1,
  	"phase_shift_tolerance": 0.1,
	"computed_channels": [1, 18, 37, 56, 75] 
  }
since I am setting the grid_size to 75e9 for 400G signal.

Besides, is there a position in the code I can plot the spectrum with tilt?

Thanks.

**Reply by jktjkt** [reply]
I think that you should definitely change `wdm_grid_size`. Either way, you're exploring beyond what we verified in the lab, so we cannot vouch that the results are going to be accurate.

> Besides, is there a position in the code I can plot the spectrum with tilt?

There's the `--show-channels` option for `gnpy-transmission-example`, perhaps it will help you.

**Reply by ojnas** [reply]
This doesn't really help if there is a ROADM at the end of the path since the power levels will be equalized in that case.

**Reply by caffery-chen** [reply]
Hi @ojnas ,

this reminds me of the ROADM setting in eqpt_config.json.

      "Roadm":[{
            "target_pch_out_db": -20,
            "add_drop_osnr": 38,
            "pmd": 0,
            "restrictions": {
                            "preamp_variety_list":[],
                            "booster_variety_list":[]
                            }            
            }],

target_pch_out_db is well understood that all channel power is equalized to -20 dBm at the output of ROADM.
The add_drop_osnr is a little confusing, because in the doc, this parameter defines the OSNR penalty due to add/drop. But 38 dB OSNR penalty...shouldn't be here with some smaller value like <2 dB degradation due to ROADM neighboring channel leakage.

**Reply by ojnas** [reply]
The add_drop_osnr is not for modeling channel leakage but for modeling the ASE contribution from any amplifiers in the add/drop stages of the ROADMs. And it's not a penalty but the OSNR resulting from this contribution only. It is combined with the OSNR after propagation here:
https://github.com/Telecominfraproject/oopt-gnpy/blob/924c56850d77c53f8e5c46e21082e5510a8b00d5/gnpy/topology/request.py#L343
where the update_snr method performs "SNR summing":
https://github.com/Telecominfraproject/oopt-gnpy/blob/924c56850d77c53f8e5c46e21082e5510a8b00d5/gnpy/core/elements.py#L119

**Reply by caffery-chen** [reply]
> In my recent work, the SRS can be taken into account also in the `Fiber` instance, but it is not the default. In order to enable this feature, the Raman flag within the simulation parameters has to be set as `True`.

Hi @AndreaDAmico ,

I have done so, but I do not see how the raman effect is included in the propagation func in Fiber. Not sure where actually this SRS is included.

I checked the "issues" and it claim that this SRS model includes also the WDL, does this still hold?

Thanks

**Reply by AndreaDAmico** [reply]
Dear @caffery-chen ,
my recent work has not been merged in the master yet. As @jktjkt pointed out, using the current version you are forced to use the `RamanFiber` in order to include the SRS effect. The previous messages regarding the simulation parameters are still valid.

**Reply by caffery-chen** [reply]
> The add_drop_osnr is not for modeling channel leakage but for modeling the ASE contribution from any amplifiers in the add/drop stages of the ROADMs. And it's not a penalty but the OSNR resulting from this contribution only. It is combined with the OSNR after propagation here:
> https://github.com/Telecominfraproject/oopt-gnpy/blob/924c56850d77c53f8e5c46e21082e5510a8b00d5/gnpy/topology/request.py#L343
> 
> 
> where the update_snr method performs "SNR summing":
> https://github.com/Telecominfraproject/oopt-gnpy/blob/924c56850d77c53f8e5c46e21082e5510a8b00d5/gnpy/core/elements.py#L119

Dear @ojnas,

The update_snr is called in the last node. If I passed through 3 ROADM nodes, I should do the following right?

add_drop_osnr = equipment['Roadm']['default'].add_drop_osnr
path[-1].update_snr(req.tx_osnr, add_drop_osnr ,add_drop_osnr ,add_drop_osnr)

Thanks.

**Reply by jktjkt** [reply]
No, the Add/Drop OSNR penalty is the incremental OSNR penalty for both the Add stage (at the very beginning of the path) and the final Drop stage. In the intermediate ROADM nodes, the signals do not cross the Add/Drop stages, they go via the *express* paths and therefore the `add_drop_osnr` does not come into consideration.
